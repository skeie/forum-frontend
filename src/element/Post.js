// @flow

import React from 'react';
import { View } from 'react-native';
import Hyperlink from 'react-native-hyperlink';
import { Text } from '../common';

import type { Content } from '../types';

type Props = {
    content: Content,
};

const Post = ({ content }: Props) => {
    return (
        <View style={{ width: '100%' }}>
            {content.map((content, index) => (
                <Hyperlink
                    linkDefault
                    key={`${content}-${index}`}
                    linkStyle={{ color: '#C0DEEF' }}>
                    <Text
                        style={{
                            marginTop: index > 0 ? 10 : 0,
                            color: '#B6B9BB',
                        }}>
                        {content}
                    </Text>
                </Hyperlink>
            ))}
        </View>
    );
};

export default Post;
