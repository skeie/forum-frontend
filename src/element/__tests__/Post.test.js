import React from 'react';
import Post from '../Post';
import posts from '../../testData/posts';
import renderer from 'react-test-renderer';

const content = ['', 'sapdap dette er fett', ''];

it('Posts renders correctly', () => {
    const tree = renderer.create(<Post content={content} />).toJSON();
    expect(tree).toMatchSnapshot();
});
