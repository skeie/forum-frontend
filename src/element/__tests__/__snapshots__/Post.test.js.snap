// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts renders correctly 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    linkDefault={true}
    linkStyle={
      Object {
        "color": "#C0DEEF",
      }
    }
    onPress={[Function]}
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "poppins",
          },
          null,
          Object {
            "color": "#B6B9BB",
            "marginTop": 0,
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    linkDefault={true}
    linkStyle={
      Object {
        "color": "#C0DEEF",
      }
    }
    onPress={[Function]}
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "poppins",
          },
          null,
          Object {
            "color": "#B6B9BB",
            "marginTop": 10,
          },
        ]
      }
    >
      sapdap dette er fett
    </Text>
  </View>
  <View
    linkDefault={true}
    linkStyle={
      Object {
        "color": "#C0DEEF",
      }
    }
    onPress={[Function]}
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "poppins",
          },
          null,
          Object {
            "color": "#B6B9BB",
            "marginTop": 10,
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
